src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoData.cs

```diff
@@ -93,5 +97,22 @@ internal static int[] GetCurrencyNegativePatterns(CultureInfo cultureInfo)
 
             throw DateTimeFormatInfoData.GetCultureNotSupportedException(cultureInfo);
         }
+        
+        private static Version GetOSXKernelVersion()
+        {
+            if (s_isOSX)
+            {
+                ulong bytesLength = 256;
+                byte[] bytes = new byte[bytesLength];
+                sysctlbyname("kern.osrelease", bytes, ref bytesLength, null, 0);
+                string versionString = Encoding.UTF8.GetString(bytes);
+                return Version.Parse(versionString);
+            }
+            
+            return new Version(0, 0, 0);
+        }
+        
+        [DllImport("libc")]
+        private static extern int sysctlbyname(string ctlName, byte[] oldp, ref ulong oldpLen, byte[] newp, ulong newpLen);
```

Comment:
> The lengths are size_t, so would oldpLen and newpLen be better as ref IntPtr and IntPtr?

Since this code is OSX specific, I thought the ABI was already constant? Which is why we don't need shims for OSX specific native calls in the product.

> And should this be SetLastError = true as well?

Added.

> Also, I wonder if we're going to run into this in other places, and if so whether we should centralize this check somewhere to make it available to other tests in the future.

I like using the rule "2nd time pays for generality".  So if/when we need it somewhere else, we can refactor it there at that time.

> It'd also be nice if we could avoid needing such a P/Invoke in the tests, and instead have such version information exposed from RuntimeInformation

#4334 adds a RuntimeInformation.OSDescription string, which will look something like `Darwin Kernel Version 15.0.0: Sat Sep 19 15:53:46`, which could be parsed by this test (once the API is added).  However, I didn't think it would be good to rely on the structure of this string.